### ISSUE TITLE
Error spawn snapshot/... ENOENT; trying to use imagemin plugin

### ISSUE THREAD

**Issue by Manusan42** [zeit/pkg#282]
Hi, 
I'm a bit lost, thus asking for help.

I try to use the "imagemin-jpeg-recompress"-module:
```typescript
import * as imageminJpegRecompress from 'imagemin-jpeg-recompress';
...
imageminJpegRecompress({
  // options...
})(buffer).then((buffer: Buffer) => /* do other stuff */)
```
If I start the app and the buffer (image data) triggers the recompress module, I get  runtime errors...

If i do not set assets for config, then I get:
```bash
Error: { Error: The system cannot find the path specified.

    at notFoundError (E:\snapshot\img-sharp\node_modules\cross-spawn\lib\enoent.js:11:11)
    at verifyENOENT (E:\snapshot\img-sharp\node_modules\cross-spawn\lib\enoent.js:46:16)
    at ChildProcess.cp.emit (E:\snapshot\img-sharp\node_modules\cross-spawn\lib\enoent.js:33:19)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn E:\\snapshot\\img-sharp\\node_modules\\jpeg-recompress-bin\\vendor\\jpeg-recompress.exe',
  killed: false,
  stdout: '',
  stderr: 'The system cannot find the path specified.\r\n',
  failed: true,
  signal: null,
  cmd: 'E:\\snapshot\\img-sharp\\node_modules\\jpeg-recompress-bin\\vendor\\jpeg-recompress.exe --quiet --accurate --quality medium --method smallfry --min 60 --max 90 --loops 6 --subsample default --strip C:\\Users\\Manusan\\AppData\\Local\\Temp\\8dc183e6-aa2b-4315-a15b-32e18e1b0bf7 C:\\Users\\Manusan\\AppData\\Local\\Temp\\6214ebcb-4c5c-409d-8291-3a45538a7a7a',
  timedOut: false }
```

If I set in package.json:
```json
  "pkg": {
    "assets": "node_modules/jpeg-recompress-bin/vendor/*"
  },
```
then I get:
```bash
Error: { Error: spawn E:\snapshot\img-sharp\node_modules\jpeg-recompress-bin\vendor\jpeg-recompress.exe ENOENT
    at exports._errnoException (util.js:1026:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:189:19)
    at onErrorNT (internal/child_process.js:366:16)
    at _combinedTickCallback (internal/process/next_tick.js:102:11)
    at process._tickCallback (internal/process/next_tick.js:161:9)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn E:\\snapshot\\img-sharp\\node_modules\\jpeg-recompress-bin\\vendor\\jpeg-recompress.exe',
  path: 'E:\\snapshot\\img-sharp\\node_modules\\jpeg-recompress-bin\\vendor\\jpeg-recompress.exe',
  spawnargs:
   [ '--quiet',
     '--accurate',
     '--quality',
     'medium',
     '--method',
     'smallfry',
     '--min',
     60,
     '--max',
     90,
     '--loops',
     6,
     '--subsample',
     'default',
     '--strip',
     'C:\\Users\\Manusan\\AppData\\Local\\Temp\\aa432eda-0f8d-4d2f-be06-91e4728477de',
     'C:\\Users\\Manusan\\AppData\\Local\\Temp\\a78a387c-8c0c-4a95-b388-a78cf8174411' ],
  killed: false,
  stdout: '',
  stderr: '',
  failed: true,
  signal: null,
  cmd: 'E:\\snapshot\\img-sharp\\node_modules\\jpeg-recompress-bin\\vendor\\jpeg-recompress.exe --quiet --accurate --quality medium --method smallfry --min 60 --max 90 --loops 6 --subsample default --strip C:\\Users\\Manusan\\AppData\\Local\\Temp\\aa432eda-0f8d-4d2f-be06-91e4728477de C:\\Users\\Manusan\\AppData\\Local\\Temp\\a78a387c-8c0c-4a95-b388-a78cf8174411',
  timedOut: false }
```

pkg compile output has no warnings for this module.

I have no clue what I can try or if it's just not possible to use the module?
Any help is appreciated. Thanks.

**Comment by Manusan42** [comment]
Ok, nevermind,
after reading some responses to other tickets, I'm quite sure this is not solvabale by using the imagemin-jpeg-recompress module which internally spawns the child process, because pkg's snapshot-system automatically rewrites the access path.

But lukily the mudule is simplistic and I copied the code into my project. Then I was able to copy the binary to spawn with my build process to the destination folder next to the compiled application executable and using:
```typescript
const binPath = path.join(path.parse(process.execPath).dir, path.parse(origBinPath).base);
```
to access the binary directly.

**Comment by Jucesr** [comment]
Hi @Manusan42 

I am having the same problem with the module [html-pdf](https://www.npmjs.com/package/html-pdf).

Did you modify the code of the module?

**Comment by Manusan42** [comment]
It's been a while since I worked on that project and I do not remember, but I looked just quickly into the source and it seems that I have used the modules code and just mixed it with my code into one file.
Of course the module had dependencies which I just put to my imports.

It was a small project and a small module, so there was no big problem figuring out how to use the code directly.

I wish you good luck with your project.

Best,
Manuel
